cpp/tests/dictionary/gather_test.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/dictionary/dictionary_column_view.hpp>
+#include <cudf/dictionary/encode.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/sorting.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+#include <vector>
+
+struct DictionaryGatherTest : public cudf::test::BaseFixture {};
+
+TEST_F(DictionaryGatherTest, Gather)
+{
+    cudf::test::strings_column_wrapper strings{ "eee", "aaa", "ddd", "bbb", "ccc", "ccc", "ccc", "eee", "aaa" };
+    auto dictionary = cudf::dictionary::encode( strings );
```

Comment:
This isn't specifically related to this PR, but why isn't this function named like the usual factory functions (make_fixed_width_column, etc)?

Reply:
We decided that this is not really a factory since it converts a set of data from a `column_view` . 
Here is the comment/discussion in the original PR:
https://github.com/rapidsai/cudf/pull/3577#discussion_r366980877